controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -721,6 +729,95 @@ func (r *StorageConsumerReconciler) reconcileNoobaaAccount() error {
 	return nil
 }
 
+func (r *StorageConsumerReconciler) deletionPhase() (reconcile.Result, error) {
+
+	consumerConfigMap := &corev1.ConfigMap{}
+	consumerConfigMap.Namespace = r.namespace
+	consumerConfigMap.Name = cmp.Or(
+		r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name,
+		fmt.Sprintf("storageconsumer-%v", util.FnvHash(r.storageConsumer.Name)),
+	)
+	if err := r.get(consumerConfigMap); client.IgnoreNotFound(err) != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Get config map's controller reference
+	controllerIndex := slices.IndexFunc(
+		consumerConfigMap.OwnerReferences,
+		func(ref metav1.OwnerReference) bool { return ptr.Deref(ref.Controller, false) },
+	)
+	var controllerRef *metav1.OwnerReference
+	if controllerIndex != -1 {
+		controllerRef = &consumerConfigMap.OwnerReferences[controllerIndex]
+	}
```

Comment:
If the configmap has no controller ref, how does that effect the deletion process? should we continue with the procedure? should we assume we are the controller? should we fail the reconcile?